### DISCUSSION TITLE
[Feature]: Be notified when pictures are added in album

### DISCUSSION THREAD

**Discussion by axgd-code** [immich-app/immich#1671] [Feature Request]
### Feature detail

In webapp and mobile app : 

- As the owner of an album, I would like to be notify when someone add pictures.
- As a viewer of an album, I would like to be notify when someone add pictures

I think that a rss feed could be a simple solution. 
A webhook could be a better solution to be notified or to integrate with n8n or IFTTT.

### Platform

Server

**Comment by feerlessleadr** [comment]
This is really the last thing that is missing for me to replace Google photos features. Looking forward to it!

**Comment by pman07** [comment]
If anyone is interested, I setup a docker container that runs a python script every 15 minutes checking user defined albumIDs for new items. If there are new items it sends a push notification via ntfy (can be self hosted). Just thought I'd share. Source code and setup details below.

https://github.com/pman07/Immich_Notify

**Reply by mjsamberg** [reply]
This is awesome! I'd love to see a native-type implementation that's "grandma proof", but this is a great first step.

**Reply by feerlessleadr** [reply]
@pman07 Thanks for this. I'm having an issue getting this to run and opened up an issue on your repo, but tagging you here in case you didn't see:

https://github.com/pman07/Immich_Notify/issues/2

**Comment by Godd67** [comment]
This would be awesome! Desperately need this feature. Use case - have an album shared with family members. If one of us decided to share a photo, others should be notified automatically.

**Reply by feerlessleadr** [reply]
I would highly recommend spinning up immich-notify (https://github.com/pman07/Immich_Notify).

Works great for notification on new assets added to albums.

Can't speak to iOS, but ntfy on Android is awesome.

**Reply by pman07** [reply]
Thanks for the shout-out @feerlessleadr. Unless things have changed last I heard they don't see a path towards a built in notification feature. 

I've also been a fan of NTFY, this got me using it for other projects too.

**Reply by locqust** [reply]
How does this work with multiple users? I'm assuming you can only use one API key from a single user and it watches any shared albums you define no matter which user owns the albums?

**Reply by pman07** [reply]
Yes that's how it works. It doesn't specify *who* uploaded the pictures, just how many have been added in the last 15 minutes.

If you have some albums that you don't have access to, you could spin up another notify container with a different users API key I suppose.

**Reply by locqust** [reply]
> Yes that's how it works. It doesn't specify *who* uploaded the pictures, just how many have been added in the last 15 minutes.
> 
> If you have some albums that you don't have access to, you could spin up another notify container with a different users API key I suppose. 

Perfect, it's just shared albums between me and the wife so that's just what I need.

**Reply by fcastilloec** [reply]
@pman07 you mentioned that there's no path towards a built-in notification feature, but I couldn't find any mention of this anywhere. Do you know if this has changed? Having an additional app for notifications won't work, as some of you already mentioned, because there are always non-tech people in the family who even struggle with intuitive apps like this one.

**Reply by pman07** [reply]
I'm not sure, I honestly haven't checked up recently. Below is a screenshot from last June on discord, but not sure if anything has changed. I think I've seen some chatter about possible email notifications.

![image](https://github.com/immich-app/immich/assets/56171396/6539167b-f2f0-4977-9b0b-3e4b73d7e2c4)

**Comment by benjamonnguyen** [comment]
The multi-user feature sets Immich apart from alternatives like PhotoPrism, and personally the joy of sharing photos with friends and family is a big part of why I self-host Immich!

**Comment by lieberpi** [comment]
Would it be possible to have a "Notifications" option in the settings where one could select which events would send a notification?

For example:

    An album has been shared with you
    A new picture/video has been added to an album that has been shared with you
    A new comment has been made to a picture in an album that has been shared with you
    Someone liked an image in an album that has been shared with you

This would be great!

**Comment by BourbonDoc** [comment]
Just want to add that I'm extremely interested in this feature. The program itself is phenomenal, and I'm basically this feature away from self-hosting our entire family's photos. The python script, while seemingly an awesome workaround, won't work for the grandparents!

**Comment by Thinkscape** [comment]
Integration with `ntfy` would be amazing, since they are natively self-hosted (and offer cloud version if needed).

https://docs.ntfy.sh/develop/?h=ios#ios-app

**Reply by feerlessleadr** [reply]
You can already do this with this container 

https://github.com/pman07/Immich_Notify

I use it and it works great.

**Reply by Thinkscape** [reply]
Requires a 3rd party app for actually receiving them - I won't have much luck getting my family installing and manually configuring it.

A proper integration would send push notifications from the Immich app.

**Comment by kintaro1981** [comment]
Maybe https://github.com/automatisch/automatisch can help?

**Comment by aero31aero** [comment]
Why is this issue closed?

**Reply by bo0tzz** [reply]
https://immich.app/docs/administration/email-notification

**Reply by Thinkscape** [reply]
Email notifications landed, but this feature request is for other notifications types, like push/rss/webhook/ntfy/shoutrrr.

@axgdcode do you mind renaming this ticket so it reads more like
```
[Feature]: Push/webhook/etc notifications when items added to album
```

**Reply by Thinkscape** [reply]
@bo0tzz could you rename and reopen this ticket?

**Reply by axgd-code** [reply]
@Thinkscape, I think it would be easier to open a new issue. The notification system already exists with email. The new issue could be: 'Expand the notification system with webhooks or push providers (e.g., ntfy).

**Reply by aero31aero** [reply]
Actually, my question was two-fold. Maybe its the Github UI or something else, but on issues, we clearly see who closed this issue, and why (marked-as-resolved or something), but exploring closed discussions is an opaque mystery on what was the end resolution...

**Reply by melyux** [reply]
Who's using email notifications in 2025? Ridiculous

**Reply by feerlessleadr** [reply]
I wish we could have a notification directly from the app, similar to Google photos. My family wants to be able to click a notification and have the app open right up in their phone.

**Reply by Kernald** [reply]
@melyux lots of people.

**Reply by Thinkscape** [reply]
Continues in #18120

**Comment by miguelangel-nubla** [comment]
https://github.com/immich-app/immich/discussions/13181

**Comment by aslmx** [comment]
Whats the status of this? 

I want to come back to the intials requests aspect of using RSS

> I think that a rss feed could be a simple solution.

I also think RSS is the right thing for this. Alternatively Atom can be used.

While i selfhost Gotify myelf (which is similar to ntfy but was there before ntfy, which is also cool), it would be cool if Immich would allow a native rss or atom feed that would just be there.
Github allows you to just add `.atom` to e.g. the release page and give you a feed about a projects releases. (I just remembered that). 

My Usecase or the reason I was looking for this feature is the following:
I setup an Immich instance and a friend of mine also setup one instance and now he shared me an Album from his vacation. 
I use an RSS reader all day to collect various news and It would be easy, to just add another RSS feed.

I would imagine the Usability of this as follows

Album creator:
- The User creating a shared album can activate a toggle "Allow users to follow Albums RSS Feed"

Album User:
- Any one having the link to an album will find a link to the RSS feed on the Album page. Optionally there will be the RSS icon (see [RSS Wikipedia Article](https://en.wikipedia.org/wiki/RSS)) and a click copies the RSS URL into the users Clipboard.
- I don't know if there is some specified `rss://` URI scheme that will launch the users favourite RSS reader and lets him add a feed with one click, but if it exists, this could also be the default click action.
- Alternatively, one could just add `.rss` or `.atom` to the link and Immich would understand that.

With rss and atom being established standards for this, I would think there are many libraries that allow for an easy implementation of such a feature without much hassle.

I would very much appreciate this!

**Comment by caseycarlson11** [comment]
I would like to set up shared albums where only the owner of the album would receive an email notification. This way everyone in the office could have their own album which acts like a dropbox with an email notifier. Currently all participants need to be sharing the album as an editor to be able to send pictures to each other. Notifying everyone when all I want to do is notify the owner is not gonna work for my use case. 

Would be nice if the email template had a way to tell you who put the photo in the shared folder. Having this info in the subject of the email would be nice so that you know you got a photo added and by who without opening the email.

**Comment by kingp0dd** [comment]
+1
